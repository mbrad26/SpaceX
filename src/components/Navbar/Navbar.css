nav {
  background-color: black;
}

img {
  display: block;
  width: 100%;
  height: auto;
}
