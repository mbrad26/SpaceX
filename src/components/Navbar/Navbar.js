import React, { useContext } from 'react';
import logo from '../../assets/SpaceX-Logo.wine.svg';
import Context from '../../context/context.js';
import './NavBar.css';

import { Navbar, Container, Image } from "react-bootstrap";

const NavBar = () => {
  const { handleClick } = useContext(Context);
  const handleRefresh = () => window.location.reload();

  return (
    <Navbar expand="lg">
      <Container className="links-container">
        <button onClick={handleRefresh}>
          <Image
            src={logo}
            className="d-inline-block align-top"
            alt=""
            loading="lazy"
          />
        </button>
        <ul className="navbar-nav mr-auto">
          <li className="nav-item active">
            <button className="nav-link" onClick={handleClick}>ROCKETS</button>
          </li>
          <li className="nav-item">
            <button className="nav-link" onClick={handleClick}>DRAGONS</button>
          </li>
        </ul>
      </Container>
    </Navbar>
  )
};

export default NavBar;
