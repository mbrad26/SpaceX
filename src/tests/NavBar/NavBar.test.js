import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Provider from '../../context/provider';
import Navbar from '../../components/NavBar/NavBar';
import Context from '../../context/context';

describe('NavBar', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider>
        <Navbar />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls handleRefresh', () => {
    // const { location } = window;
    delete window.location;
    window.location = { reload: jest.fn() };

    const context = {
      handleClick: jest.fn()
    }
    const { container } = render(
      <Provider value={context}>
        <Navbar />
      </Provider>
    );

    fireEvent.click(screen.getAllByRole('button')[0]);

    expect(window.location.reload).toHaveBeenCalledTimes(1);
  });
});
