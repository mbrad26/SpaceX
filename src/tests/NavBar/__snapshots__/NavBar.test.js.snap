// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders snapshot 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-light"
>
  <div
    class="links-container container"
  >
    <button>
      <img
        alt=""
        class="d-inline-block align-top"
        loading="lazy"
        src="SpaceX-Logo.svg"
      />
    </button>
    <ul
      class="navbar-nav w-100"
    >
      <li
        class="nav-item active"
      >
        <button
          class="nav-link ml-auto"
        >
          ROCKETS
        </button>
      </li>
      <li
        class="nav-item"
      >
        <button
          class="nav-link ml-auto"
        >
          DRAGONS
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
